### DISCUSSION TITLE
Wire:model Data Binding Failed to Work

### DISCUSSION THREAD

**Discussion by keithyeohh** [livewire/livewire#5639] [1. Help]
I am trying out Laravel Livewire version **2.12**, and when I attempted to make use of `wire:model` it doesn't seems to be working. Following are my codes.

The expected results which I am expecting the `<h1>` displaying **Livewire**, then while I am typing on the text field, it should reflect accordingly on the `<h1>` tag. Also, I am not seeing any console errors from my Google Chrome.

## Login.php
```
<?php

namespace App\Http\Livewire\Auth;

use Livewire\Component;

class Login extends Component
{
    public $message = 'Livewire';

    public function render()
    {
        return view('livewire.auth.login');
    }
}
```

## app.blade.php
```
<!DOCTYPE html>
<html>
<head>
    <title>My Livewire Page</title>
    <livewire:styles />
</head>
<body>
    @yield('content')

    <livewire:scripts />
</body>
</html>
```

## login.blade.php
```
@extends('layouts.app')

@section('content')

<div>
    {{ $message }}

    <input type="text" wire:model="message">
</div>

@endsection
```

**Comment by Martleby** [comment]
I think you want:

```blade
@livewireStyles
````

and

```blade
@livewireScripts
```

in your template.

**Reply by joshhanley** [reply]
@Martleby they have `<livewire:styles />` and `<livewire:scripts />` which does the same thing

**Comment by joshhanley** [comment]
@keithyeohh the issue is you're using the old [layouts using template inheritance](https://laravel.com/docs/10.x/blade#layouts-using-template-inheritance), where as Livewire by default works with [layouts using components](https://laravel.com/docs/10.x/blade#layouts-using-components).

So you can either switch to using the layout component style if you want it to work by default, otherwise, you need to tell Livewire that you are using the old style by using `->extends()` in your render method, see here for details https://laravel-livewire.com/docs/2.x/rendering-components#custom-layout

Hope this helps!
